certbot_dns_loopia.py

```diff
@@ -57,27 +62,39 @@ def _get_loopia_client(self):
 
     def _perform(self, domain, validation_name, validation):
         loopia = self._get_loopia_client()
-        domain_parts = split_domain(validation_name)
+        domain_parts = self._tld_extract(validation_name)
 
         dns_record = DnsRecord("TXT", ttl=self.ttl, data=validation)
 
         msg = "Creating TXT record for %s on subdomain %s"
-        logger.debug(msg, *domain_parts)
+        logger.debug(msg, domain_parts.registered_domain, domain_parts.subdomain)
 
-        loopia.add_zone_record(dns_record, *domain_parts)
+        loopia.add_zone_record(
+            record=dns_record,
+            domain=domain_parts.registered_domain,
+            subdomain=domain_parts.subdomain or None,
```

Comment:
For those wondering: the `or None` is just a hack to convert `domain_parts.subdomain` from an empty string to `None` when needed, since `loopialib` expects an explicit `None` and not an empty string, which is what `tldextract` generates when there is no subdomain.